ios/brave-ios/Sources/Onboarding/WelcomeFocus/FocusStepsScreenView.swift

```diff
@@ -0,0 +1,270 @@
+        .clipShape(RoundedRectangle(cornerRadius: 16.0))
+        .tabViewStyle(PageTabViewStyle(indexDisplayMode: .never))
+        .animation(.easeInOut, value: indicatorIndex)
+        .transition(.slide)
+        .disabled(indicatorIndex > 0)
+        .padding(.bottom, 24)
+
+        Spacer()
+
+        Button(
+          action: {
+            if indicatorIndex > 0 {
+              isP3AViewPresented = true
+            } else {
+              indicatorIndex += 1
+            }
+          },
+          label: {
+            Text(Strings.FocusOnboarding.continueButtonTitle)
+              .font(.body.weight(.semibold))
+              .foregroundColor(Color(.white))
+              .dynamicTypeSize(dynamicTypeRange)
+              .padding()
+              .foregroundStyle(.white)
+              .frame(maxWidth: .infinity)
+              .background(Color(braveSystemName: .buttonBackground))
+          }
+        )
+        .clipShape(RoundedRectangle(cornerRadius: 12.0))
+        .overlay(RoundedRectangle(cornerRadius: 12.0).strokeBorder(Color.black.opacity(0.2)))
+      }
+      .opacity(opacity)
+      .onAppear {
+        withAnimation(.easeInOut(duration: 1.5).delay(1.25)) {
+          opacity = 1.0
+        }
+      }
+      .onChange(of: indicatorIndex) { newValue in
+        if indicatorIndex == 1 {
+          Timer.scheduledTimer(withTimeInterval: 3, repeats: false) { _ in
+            UIImpactFeedbackGenerator(style: .medium).vibrate()
+          }
+        }
+      }
+    }
+    .padding(.vertical, shouldUseExtendedDesign ? 64 : 20)
+    .padding(.horizontal, shouldUseExtendedDesign ? 60 : 20)
+  }
+
+  private var cancelButton: some View {
+    Button(
+      action: {
+        isP3AViewPresented = true
+      },
+      label: {
+        Image("focus-icon-close", bundle: .module)
+          .resizable()
+      }
+    )
+  }
+}
+
+struct FocusStepsHeaderTitleView: View {
+  @Binding var activeIndex: Int
+
+  let dynamicTypeRange = (...DynamicTypeSize.large)
+
+  var body: some View {
+    let title =
+      activeIndex == 0
+      ? Strings.FocusOnboarding.movingAdsScreenTitle
+      : Strings.FocusOnboarding.noVideoAdsScreenTitle
+
+    let description =
+      activeIndex == 0
+      ? Strings.FocusOnboarding.movingAdsScreenDescription
+      : Strings.FocusOnboarding.noVideoAdsScreenDescription
+
+    VStack(spacing: 10) {
+      Text(title)
+        .font(
+          Font.custom("FlechaM-Medium", size: 32)
+        )
+        .dynamicTypeSize(dynamicTypeRange)
+        .fixedSize(horizontal: false, vertical: true)
+        .opacity(0.9)
+
+      Text(description)
+        .font(
+          Font.custom("Poppins-Medium", size: 17)
+        )
+        .dynamicTypeSize(dynamicTypeRange)
+        .fixedSize(horizontal: false, vertical: true)
+        .multilineTextAlignment(.center)
+        .foregroundColor(Color(braveSystemName: .textTertiary))
+    }
+  }
+}
+
+struct FocusStepsPagingIndicator: View {
```

Comment:
private

Reply:
This is used in multiple screens but I made the Header View private